### PR TITLE
Added unit test cases for data table sorting

### PR DESCRIPTION
**Pull request by hemantk1325**
## Description

Adding unit test cases for modus data table sorting feature.

References #1169 

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: stencil-workspace/src/components/modus-data-table/modus-data-table.e2e.ts [hunk 1]
```diff
@@ -1,11 +1,247 @@
...
+    ]);
+    component.setProperty('data', [{ firstName: 'Gordon' }]);
+
+    await page.waitForChanges();
+    row = await page.findAll('modus-data-table >>> td');
+    expect(row.length).toBe(1);
+    expect(row[0].textContent).toBe('Gordon');
+  });
+
+  it('renders with primitive columns and data', async () => {
```
Comment (msankaran0712) [inline stencil-workspace/src/components/modus-data-table/modus-data-table.e2e.ts]: We don't have to write test cases for "rendering with primitive columns and data". According to our column structure, `accessorKey` is a mandatory property. Though tanstack supports, we do not officially support primitive columns and data.
please remove all the primitive data/column test cases

File: stencil-workspace/src/components/modus-data-table/modus-data-table.e2e.ts [hunk 2]
```diff
@@ -1,11 +1,247 @@
...
+        progress: 54,
+        status: 'single',
+      },
+    ]);
+    component.setProperty('sort', true);
+    await page.waitForChanges();
+    const sortEvent = await page.spyOnEvent('sortChange');
+
+    const header = await page.find('modus-data-table >>> .sort-icons');
+    await header.click();
```
Comment (msankaran0712) [inline stencil-workspace/src/components/modus-data-table/modus-data-table.e2e.ts:142]: I think validating one-click event is sufficient, though you can use multiple clicks to validate the tooltip message on each click.

File: stencil-workspace/src/components/modus-data-table/modus-data-table.e2e.ts [hunk 3]
```diff
@@ -1,11 +1,247 @@
...
+    // Descending sort
+    tableData = await page.findAll('modus-data-table >>> td');
+    tableDataTexts = await Promise.all(tableData.map((td) => td.textContent));
+
+    expect(tableDataTexts[0]).toEqual('Elliott');
+    expect(tableDataTexts[1]).toEqual('Bob');
+    expect(tableDataTexts[2]).toEqual('Agnes');
+
+    expect(sortEvent).toHaveReceivedEventTimes(2);
+  });
```
Comment (msankaran0712) [inline stencil-workspace/src/components/modus-data-table/modus-data-table.e2e.ts:198]: - Please add a test case to verify if the sort is false.
- Add a test case for verifying `showSortIconOnHover`
- Add a test case to verify different sorting messages 
- Add a test case to verify sorting on numerical column

File: stencil-workspace/src/components/modus-data-table/modus-data-table.e2e.ts [hunk 4]
```diff
@@ -1,11 +1,247 @@
...
+
+    const header = await page.find('modus-data-table >>> .sort-icons');
+    await header.click();
+    await header.click();
+    await header.click();
+
+    expect(sortEvent).toHaveReceivedEventTimes(3);
+  });
+
+  it('should output sort data on sort icon click with sort enabled', async () => {
```
Comment (msankaran0712) [inline stencil-workspace/src/components/modus-data-table/modus-data-table.e2e.ts:149]: verify sorting descending order too

**Review by msankaran0712 — CHANGES_REQUESTED** [review]
- Verify ascending and descending sort icons are rendered in the correct order

File: stencil-workspace/src/components/modus-data-table/modus-data-table.e2e.ts (hunk 2 above)
Comment (msankaran0712) [reply stencil-workspace/src/components/modus-data-table/modus-data-table.e2e.ts:142]: In one test case, you can verify both the sort icon and tooltip message.

File: stencil-workspace/src/components/modus-data-table/modus-data-table.e2e.ts [hunk 5]
```diff
@@ -1,11 +1,358 @@
...
+
+    await page.waitForChanges();
+
+    const sortContainer = await page.find('modus-data-table >>> .can-sort');
+    expect(sortContainer).not.toBeNull();
+  });
+
+  it('should output sort event on sort icon click with sort enabled', async () => {
+    const component = await page.find('modus-data-table');
+    component.setProperty('columns', [
```
Comment (Ankur-Koul) [inline stencil-workspace/src/components/modus-data-table/modus-data-table.e2e.ts:120]: Put this data in a constant variable and then use it.

**Comment by hemantk1325** [comment]
closing this PR. Raised a new PR to `modus-table-main` branch
https://github.com/trimble-oss/modus-web-components/pull/1462

**Pull request closed without merging**